test.py

```diff
@@ -0,0 +1,157 @@
+import json
+import logging
+import types
+import webapp2
+from datetime import datetime, timedelta
+
+import persistentcache
+from google.appengine.api import memcache
+from google.appengine.ext import db
+
+
+TEST1_KEY = "test1"
+TEST2_KEY = "test2"
+
+# 2 megabyte payload
+# (uncomment if you can figure out how to work around appengine's inability to store huge entities)
+#TEST2_PAYLOAD = ("0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ.." * 1000) # 31250
+
+class TestModel(db.Model):
+    payload = db.TextProperty()
+    #blobload = db.BlobProperty()
+
+    def memcache_key(self):
+        return self.payload
+
+    @staticmethod
+    def getcache(key):
+        return persistentcache.get(key)
+
+    def putcache(self, ttl=None):
+        # Send this to both persistentcache and the datastore.
+        rv = self.put()
+        mk = self.memcache_key()
+        persistentcache.set(mk, self, ttl)
+    
+    def deletecache(self):
+        mk = self.memcache_key()
+        persistentcache.delete(mk)
+        self.delete()
+        
+class CacheTestRequestHandler(webapp2.RequestHandler):
+    def report_failure(self, msg=None):
+        if msg is not None:
+            self.response.write("Error: %s\n" % msg)
+        self.response.write("Failure")
+
+    def report_exception(self, e):
+        if e is not None:
+            import traceback
+            msg = "Exception: (%s) %s\n%s" % (type(e).__name__, e.message, traceback.format_exc())
+            self.response.write(msg)
+            logging.exception(msg)
+        self.response.write("Failure")
+    
+    def report_success(self):
+        self.response.write("Success")
+
+    def get(self):
+        self.report_failure("Abstract method CacheTestRequestHandler.get called") 
+
+class SetupTest(CacheTestRequestHandler):
+    def get(self):
+        try:
+            e = TestModel(key_name=TEST1_KEY, payload=TEST1_KEY)
+            e.putcache()
+            #e = TestModel(key_name=TEST2_KEY, payload=TEST2_KEY, blobload=TEST2_PAYLOAD)
+            #e.putcache()
+            self.report_success()
+        except Exception, e:
+            self.report_exception(e)
+
+class CleanupTest(CacheTestRequestHandler):
```

Comment:
by the by, there's better ways to delete many records in GAE/DB:
https://cloud.google.com/appengine/docs/python/memcache/clientclass#Client_delete_multi
https://cloud.google.com/appengine/docs/python/memcache/clientclass#Client_delete_multi_async